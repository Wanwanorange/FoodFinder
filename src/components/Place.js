import React, { Component } from 'react';
import '../styles/Place.css';

class Place extends Component {

    removePlace = () => {
        this.props.removePlace({ id: this.props.id });
    }

    completePlace = () => {
        this.props.completePlace({ id: this.props.id });
    }

    render() {
        return (
            <div className="place">
                <div className="placeCheck">
                    <p>{this.props.name}, {this.props.category}</p>
                    {!this.props.completed &&
                        <button className="completeButton"
                            onClick={this.completePlace}>Been there!</button>}
                </div>
                <button className="removeButton"
                    onClick={this.removePlace} id={this.props.id}>X</button>
            </div>
        );
    }
}

export default Place