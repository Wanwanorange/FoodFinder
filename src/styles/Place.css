.place {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 5px 10px 5px 10px;
    border-bottom: solid 1px lightgrey;
}

.place button {
    height: 100%;
}

.place-details {
    margin: 5px;
}