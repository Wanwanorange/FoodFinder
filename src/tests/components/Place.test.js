import React from 'react';
import {shallow} from 'enzyme';
import Place from '../../components/Place';
import places from '../testData/places';

let wrapper, removePlace, completePlace;

beforeEach(() => {
    removePlace = jest.fn();
    completePlace = jest.fn();
    wrapper = shallow(<Place
        removePlace={removePlace}
        completePlace={completePlace}
        id={places[0].id}/>);
});

test('should render Place correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should remove place when prompted', () => {
    wrapper.find('.remove-button').simulate('click');
    expect(removePlace).toHaveBeenLastCalledWith({ id: '1234' });
});

test('should complete place when prompted', () => {
    wrapper.find('.complete-button').simulate('click');
    expect(completePlace).toHaveBeenCalledWith({ id: '1234' });
});
