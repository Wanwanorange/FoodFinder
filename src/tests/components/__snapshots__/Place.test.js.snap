// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Place correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Place
    completePlace={[Function]}
    id="1234"
    removePlace={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="placeCheck"
>
          <p>
                    , 
          </p>
          <button
                    className="completeButton"
                    onClick={[Function]}
          >
                    Been there!
          </button>
</div>,
        <button
          className="removeButton"
          id="1234"
          onClick={[Function]}
>
          X
</button>,
      ],
      "className": "place",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              , 
</p>,
            <button
              className="completeButton"
              onClick={[Function]}
>
              Been there!
</button>,
          ],
          "className": "placeCheck",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                ", ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              ", ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Been there!",
              "className": "completeButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Been there!",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "removeButton",
          "id": "1234",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="placeCheck"
>
            <p>
                        , 
            </p>
            <button
                        className="completeButton"
                        onClick={[Function]}
            >
                        Been there!
            </button>
</div>,
          <button
            className="removeButton"
            id="1234"
            onClick={[Function]}
>
            X
</button>,
        ],
        "className": "place",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                , 
</p>,
              <button
                className="completeButton"
                onClick={[Function]}
>
                Been there!
</button>,
            ],
            "className": "placeCheck",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  ", ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                ", ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Been there!",
                "className": "completeButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Been there!",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "removeButton",
            "id": "1234",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
